### PR TITLE
New rule `selector-max-empty-lines`.

### PR DESCRIPTION
**Pull request by evilebottnawi**

### CODE REVIEW

File: docs/user-guide/rules.md [hunk 1]
```diff
@@ -153,6 +153,7 @@ Here are all the rules within stylelint, grouped by the [*thing*](http://apps.wo
 - [`selector-pseudo-element-colon-notation`](../../src/rules/selector-pseudo-element-colon-notation/README.md): Specify single or double colon notation for applicable pseudo-elements.
 - [`selector-root-no-composition`](../../src/rules/selector-root-no-composition/README.md): Disallow the composition of `:root` in selectors.
 - [`selector-type-case`](../../src/rules/selector-type-case/README.md): Specify lowercase or uppercase for type selector.
+- [`selector-max-empty-lines`](../../src/rules/selector-max-empty-lines/README.md): Limit the number of adjacent empty lines within selector.
```
Comment (davidtheclark) [inline docs/user-guide/rules.md]: within selectors (plural)

File: src/rules/selector-max-empty-lines/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# selector-max-empty-lines
+
+Limit the number of adjacent empty lines within selector.
```
Comment (davidtheclark) [inline src/rules/selector-max-empty-lines/README.md]: within selectors (plural)

File: src/rules/selector-max-empty-lines/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+```css
+.foo,
+              /* ← */
+.bar {        /* ↑ */
+  color: red; /* ↑ */
+}             /* ↑ */
+/**              ↑
+ *          These lines */
```
Comment (davidtheclark) [inline src/rules/selector-max-empty-lines/README.md]: Let's say "This empty line"

File: src/rules/selector-max-empty-lines/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+```
+
+## Options
+
+`int`: Maximum number of characters allowed.
```
Comment (davidtheclark) [inline src/rules/selector-max-empty-lines/README.md]: number of empty lines

File: src/rules/selector-max-empty-lines/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+```css
+
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+
+```
```
Comment (davidtheclark) [inline src/rules/selector-max-empty-lines/README.md]: Did you give up on the examples?

**Comment by davidtheclark** [comment]
I think the amount of redundancy in the tests here is a little overboard. For example, I do not think it is important to repeatedly test whether the rule is accidentally checking before the selector, within the block, and after the block once it's been established that it's not. Then again, it seems like a shame to delete tests that have been written, because even though they may be redundant we definitely don't want any of them to fail. Does anybody have any concerns about the quantity of tests here?

File: src/rules/selector-max-empty-lines/README.md (hunk 5 above)
Comment (evilebottnawi) [inline src/rules/selector-max-empty-lines/README.md]: @davidtheclark I forgot while writing tests)

**Comment by evilebottnawi** [comment]
@davidtheclark @jeddy3 ping

**Comment by jeddy3** [comment]
> For example, I do not think it is important to repeatedly test whether the rule is accidentally checking before the selector, within the block, and after the block once it's been established that it's not.

Agreed. There is a fair bit of noise that makes reviewing the tests that little more time consuming.

> Then again, it seems like a shame to delete tests that have been written, because even though they may be redundant we definitely don't want any of them to fail.

Agreed also. We can leave this one as is. We should avoid redundancy going forward though.

This PR looks good to me. BTW. 

It'll need a rebase though.

**Comment by evilebottnawi** [comment]
@jeddy3 ping

**Comment by ntwb** [comment]
Can pull requests mention the issue it relates to please, it makes it easier to track for everyone 😄 

Ref #1142

**Comment by jeddy3** [comment]
> Can pull requests mention the issue it relates to please.

@evilebottnawi Please can you include a `.Closes #issueno.` in your first commit message, so this is done automatically :)

**Pull request merged**